node/adapter/datastore.py

```diff
@@ -606,6 +606,24 @@ def set_endpoint(self, hostname, container_id, endpoint):
                                    "endpoint_id": endpoint.ep_id}
         self.etcd_client.write(ep_path, endpoint.to_json())
 
+    def update_endpoint(self, hostname, container_id,
+                        old_endpoint, new_endpoint):
+        """
+        Update a single endpoint object to the datastore. Fails if the
+        old_endpoint that's passed in doesn't match what's in the datastore.
```

Comment:
It would be helpful to include an example here, e.g.

```
old_endpoint = datastore.get_endpoint(...)
new_endpoint = old_endpoint.copy()
# modify new endpoint fields
datastore.update_endpoint(..., old_endpoint, new_endpoint)
```